account_invoice_report_grouped_by_picking/models/account_move.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2017 Tecnativa - Carlos Dauden
+# Copyright 2018 Tecnativa - David Vidal
+# Copyright 2018-2019 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from collections import OrderedDict
+
+from odoo import api, models
+from odoo.tools import float_is_zero
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def _sort_grouped_lines(self, lines_dic):
+        return sorted(
+            lines_dic,
+            key=lambda x: (
+                (x["picking"].date, x["picking"].date_done or x["picking"].date)
+            ),
+        )
```

Comment:
When mixing with invoice line that doesn't have neither delivery nor sale order line, this code caused error.

```
  File "/opt/OCA/account-invoice-reporting/tree/16.0/account_invoice_report_grouped_by_picking/models/account_move.py", line 104, in lines_grouped_by_picking
    return no_picking + self._sort_grouped_lines(with_picking)
  File "/opt/OCA/account-invoice-reporting/tree/16.0/account_invoice_report_grouped_by_picking/models/account_move.py", line 18, in _sort_grouped_lines
    return sorted(
TypeError: '<' not supported between instances of 'bool' and 'datetime.datetime'

The above exception was the direct cause of the following exception:

odoo.addons.base.models.ir_qweb.QWebException: Error while render the template
TypeError: '<' not supported between instances of 'bool' and 'datetime.datetime'
Template: account.report_invoice_document
Path: /t/t/div[2]/div[1]/table/tbody/t[3]
Node: <t t-foreach="o.lines_grouped_by_picking()" t-as="lines_group"/>
```

IMHO, there should be better solution than this.
```suggestion
        DTF = '%Y-%m-%d %H:%M:%S'
        return sorted(
            lines_dic,
            key=lambda x: (
                x["picking"] and (
                    x["picking"].date and x["picking"].date.strftime(DTF),
                    (x["picking"].date_done or x["picking"].date).strftime(DTF),
                ) or ('', '')
            ),
        )
```